### PR TITLE
LPD-30589 - Rewrite Localfile.Organizations#AddOrganizationTeam as a Playwright test

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-30589

### CODE REVIEW

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts [hunk 1]
```diff
@@ -135,3 +136,54 @@ test('LPD-28908 update user information', async ({
...
+	);
+
+	apiHelpers.data.push({
+		id: `${organization.id}_test@liferay.com`,
+		type: 'organizationUserAccountAssociation',
+	});
+
+	await editOrganizationPage.createOrganizationSite(organization.name);
+
+	// await editOrganizationPage.manageSiteLink.click();
```
Comment (smottal) [inline modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts:165]: remove this comment

File: modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts [hunk 2]
```diff
@@ -135,3 +136,54 @@ test('LPD-28908 update user information', async ({
...
+		'Details',
+		'/' + organization.name
+	);
+
+	await siteConfigurationDetailsPage.allowManualMembershipManagementToggle.check();
+	await siteConfigurationDetailsPage.saveButton.click();
+
+	await waitForSuccessAlert(page);
+
+	// await applicationsMenuPage.goToSite(organization.name);
```
Comment (smottal) [inline modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts:178]: remove this

File: modules/test/playwright/fixtures/usersAndOrganizationsPagesTest.ts [hunk 3]
```diff
@@ -5,23 +5,34 @@
...
 	serviceAccountsPage: ServiceAccountsPage;
+	siteConfigurationDetailsPage: SiteConfigurationDetailsPage;
+	siteSettingsPage: SiteSettingsPage;
+	teamsPage: TeamsPage;
 	userPersonalSitePage: UserPersonalSitePage;
 	usersAndOrganizationsPage: UsersAndOrganizationsPage;
 }>({
+	applicationsMenuPage: async ({page}, use) => {
+		await use(new ApplicationsMenuPage(page));
+	},
```
Comment (smottal) [inline modules/test/playwright/fixtures/usersAndOrganizationsPagesTest.ts:35]: do we need this inside this fixture? why we cannot use its original fixture to retrieve it in the test?

File: modules/test/playwright/pages/product-navigation-control-menu-web/ProductMenuPage.ts [hunk 4]
```diff
@@ -41,6 +45,7 @@ export class ProductMenuPage {
 		});
 		this.page = page;
 		this.pagesButton = page.getByRole('menuitem', {name: 'Pages'});
+		this.peopleButton = page.getByRole('menuitem', {name: 'People'});
```
Comment (smottal) [inline modules/test/playwright/pages/product-navigation-control-menu-web/ProductMenuPage.ts:48]: is it used?

File: modules/test/playwright/pages/product-navigation-control-menu-web/ProductMenuPage.ts [hunk 5]
```diff
@@ -56,6 +61,7 @@ export class ProductMenuPage {
 			exact: true,
 			name: 'Site Settings',
 		});
+		this.teamsButton = page.getByRole('menuitem', {name: 'Teams'});
```
Comment (smottal) [inline modules/test/playwright/pages/product-navigation-control-menu-web/ProductMenuPage.ts:64]: is it used?

File: modules/test/playwright/pages/users-admin-web/EditOrganizationPage.ts [hunk 6]
```diff
@@ -48,4 +60,16 @@ export class EditOrganizationPage {
...
+	async createOrganizationSite(organizationName: string) {
+		await (
+			await this.usersAndOrganizationsPage.organizationActionsMenu(
+				organizationName
+			)
+		).click();
+		await this.organizationEditMenuItem.click();
+		await this.organizationSiteLink.click();
+		await this.createSiteToggle.check();
+		await this.organizationSiteSaveButton.click();
```
Comment (smottal) [inline modules/test/playwright/pages/users-admin-web/EditOrganizationPage.ts:73]: can you please remove this and move the instruction directly in the test? I think we will not using this so much in the future

File: modules/test/playwright/pages/users-admin-web/TeamsPage.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+import {Locator, Page} from '@playwright/test';
+
+import {getRandomInt} from '../../utils/getRandomInt';
+import {ProductMenuPage} from '../product-navigation-control-menu-web/ProductMenuPage';
+import {searchTableRowByValue} from './UsersAndOrganizationsPage';
+
+export class TeamsPage {
+	readonly newTeamButton: Locator;
+	readonly nameInput: Locator;
```
Comment (smottal) [inline modules/test/playwright/pages/users-admin-web/TeamsPage.ts:14]: this should go before (sort them)

File: modules/test/playwright/pages/users-admin-web/TeamsPage.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+	readonly teamsTable: Locator;
+	readonly teamsTableRow: (
+		colPosition: number,
+		value: string,
+		strictEqual?: boolean
+	) => Promise<{column: Locator; row: Locator}>;
+
+	constructor(page: Page) {
+		this.newTeamButton = page.getByRole('link', {name: 'Add Team'});
+		this.nameInput = page.getByPlaceholder('Name');
```
Comment (smottal) [inline modules/test/playwright/pages/users-admin-web/TeamsPage.ts:27]: same here

File: modules/test/playwright/pages/users-admin-web/TeamsPage.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	async createNewTeam(): Promise<string> {
+		const teamName = 'Team' + getRandomInt();
+
+		await this.newTeamButton.click();
+		await this.nameInput.fill(teamName);
+		await this.saveButton.click();
+
+		return teamName;
```
Comment (smottal) [inline modules/test/playwright/pages/users-admin-web/TeamsPage.ts:59]: same here, please remove the function from there and put the logic in the test

File: modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts [hunk 10]
```diff
@@ -135,3 +136,54 @@ test('LPD-28908 update user information', async ({
...
+	applicationsMenuPage,
+	editOrganizationPage,
+	page,
+	siteConfigurationDetailsPage,
+	siteSettingsPage,
+	teamsPage,
+	usersAndOrganizationsPage,
+}) => {
+	const organization = await apiHelpers.headlessAdminUser.postOrganization();
+	await usersAndOrganizationsPage.goToOrganizations();
```
Comment (smottal) [inline modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts:151]: do we need to go to the page here (before the creation of the data) or we can create everything and then move in the page?

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:playwright:nocompile

**Comment by smottal** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/5140 for conflicts

**Pull request closed without merging**